### ISSUE TITLE
No clear_devices in BestExporter

### ISSUE THREAD

**Issue by SumNeuron** [tensorflow/tensorflow#23900]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
Yes, please see this [Colab](https://colab.research.google.com/drive/1GKAqEo7qSr6kMAgxPrOrudA5eLndQ6Ub)

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): n/a
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: n/a
- TensorFlow installed from (source or binary): n/a
- TensorFlow version (use command below): 1.10+
- Python version: 3.5
- Bazel version (if compiling from source): n/a
- GCC/Compiler version (if compiling from source): n/a
- CUDA/cuDNN version: n/a
- GPU model and memory: whatever is on Colab

You can collect some of this information using our environment capture [script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with
python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

**Describe the current behavior**

One can not train a distributed-`Estimator` and export the model, then load it on a non-distributive device because if we look at the TensorFlow docs for BestExporter

```
__init__(
    name='best_exporter',
    serving_input_receiver_fn=None,
    event_file_pattern='eval/*.tfevents.*',
    compare_fn=_loss_smaller,
    assets_extra=None,
    as_text=False,
    exports_to_keep=5
)
```
it is apparent that `clear_devices` is not an option.

**Describe the expected behavior**

Let me easily export and import `Estimators`

**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.

 see this [Colab](https://colab.research.google.com/drive/1GKAqEo7qSr6kMAgxPrOrudA5eLndQ6Ub)
**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by msymp** [comment]
Hello @SumNeuron , There some good posts on Stack Overflow about Estimator export/import:
https://stackoverflow.com/questions/50612923/tensorflow-export-estimators-for-prediction
https://stackoverflow.com/questions/51537682/how-do-i-export-an-estimator-tf-estimator-dnnclassifier
We shall close this issue now. Thanks.

**Comment by SumNeuron** [comment]
@msymp I believe you misunderstood.

as stated in the question title, this is about a missing `clear_devices` flag in the export function.

Further, I would say that your two linked SO posts are not really related.

The first just uses the `model_dir` flag in the estimator constructor. 
The second either uses the `serving_input_reciever_fn` or also the `model_dir`.

Both of which lead me to believe you are missing the point, which in the issue, under "Describe current behavior" is stated as : 

> One can not train a distributed-Estimator and export the model, then load it on a non-distributive device because if we look at the TensorFlow docs for BestExporter

In short, or to recap, without the `clear_devices` flag, the current way estimators are exported, if the `model_fn` uses any `with_device` specification (as is done in the official TF documentation for distributed models), the estimator can only be loaded to a system with at least that many devices with the same names.

So please open and re-read the issue.

**Comment by SumNeuron** [comment]
@msymp seems to be relevant again but in [Keras](https://github.com/tensorflow/tensorflow/issues/33474) as sessions takes the saved the model which doesn't have clear device options

**Comment by sushreebarsa** [comment]
@SumNeuron We see that you are using old version of tensorflow , We recommend that you upgrade to 2.6.0  and let us know if the issue still persists in newer versions .Thanks!

**Comment by SumNeuron** [comment]
My apologies if this seems a bit jaded, but this issue opened almost three years ago (at the time when estimators were not more or less tossed aside in favor of Keras).  It only required a `clear_devices` flag be lifted from lower layers to higher layers.

Googling:

https://github.com/tensorflow/estimator/blob/master/tensorflow_estimator/python/estimator/exporter.py#L165-L365

no, hasn't been added

and I'm not the first to report this either: (2017)

HERE IS THE LINE
https://github.com/tensorflow/estimator/blob/a0186faff92d1502f0314e3c86a995b540fa6eb6/tensorflow_estimator/python/estimator/estimator.py#L837


https://github.com/tensorflow/tensorflow/issues/14143


https://github.com/tensorflow/tensorflow/blob/v2.6.0/tensorflow/python/saved_model/builder_impl.py#L438-L627

**Comment by sushreebarsa** [comment]
@SumNeuron Estimators are deprecated so those are not recommended for new code. Could you please have a look at [this](https://www.tensorflow.org/guide/estimator) and let us know if it helps?
Thank you!

**Comment by SumNeuron** [comment]
@sushreebarsa sure, but it also has been four years and it just required exposing an argument...
